### PR TITLE
Foundation bookmarks

### PR DESCRIPTION
**Pull request by slafleche**
Fixes some bugs with the bookmark styles and also swaps them for a new icon.

Closes: https://github.com/vanilla/vanilla/issues/10223

### CODE REVIEW

**Comment by slafleche** [comment]
@charrondev the issue though is when you click on the bookmark link, it rerenders the DOM and you lose the icon. Is it really worth working around that?

**Comment by slafleche** [comment]
I found a workaround, let me know if you have a better idea.

File: library/src/scripts/icons/common.tsx [hunk 1]
```diff
@@ -843,3 +845,28 @@ export function ResetIcon(props: { className?: string; "aria-hidden"?: areaHidde
         </svg>
     );
 }
+
+export function BookmarkIcon(props: {
+    className?: string;
+    iconProps?: IBookmarkProps;
```
Comment (charrondev) [inline library/src/scripts/icons/common.tsx:851]:
```suggestion
```

It looks like this prop is never used

File: applications/dashboard/src/scripts/compatibilityStyles/discussionStyles.ts [hunk 2]
```diff
@@ -96,4 +106,55 @@ export const discussionCSS = () => {
...
+        .Content a.Bookmarking,
+        .Content a.Bookmarked`,
+        {
+            cursor: "pointer",
+        },
+    );
+
+    cssOut(
+        `
+        .Content a.Bookmark .svgBookmark,
```
Comment (alex-mtl) [inline applications/dashboard/src/scripts/compatibilityStyles/discussionStyles.ts:137]: is it intentional extra line break here?

Comment (alex-mtl) [inline applications/dashboard/src/scripts/compatibilityStyles/discussionStyles.ts:137]: is it intentional extra line break here?
```suggestion
        `.Content a.Bookmark .svgBookmark,
```

File: applications/dashboard/src/scripts/compatibilityStyles/discussionStyles.ts [hunk 3]
```diff
@@ -96,4 +106,55 @@ export const discussionCSS = () => {
...
+            fontSize: unit(12),
+            transform: translate(`-50%`, `-50%`),
+        },
+    );
+
+    cssOut(
+        `
+        body.Discussions .DataList .Options,
+        body.Discussions .MessageList .Options
+    `,
```
Comment (alex-mtl) [inline applications/dashboard/src/scripts/compatibilityStyles/discussionStyles.ts:155]: same question?
Shouldn't we either do it all the time when start and when end the string, or do never.
Just to have it styled the same way across the project

**Pull request merged**